cmd/sonobuoy/app/worker.go

```diff
@@ -64,6 +68,21 @@ func runGather(cmd *cobra.Command, args []string) {
 	cmd.Help()
 }
 
+// sigHandler returns a channel that will receive a message after the timeout
+// elapses after a SIGTERM is received.
+func sigHandler(timeout time.Duration) <-chan struct{} {
+	stop := make(chan struct{})
+	go func() {
+		sigc := make(chan os.Signal, 1)
+		signal.Notify(sigc, syscall.SIGTERM)
+		sig := <-sigc
+		logrus.WithField("signal", sig).Info("received a signal. Waiting then sending the real shutdown signal.")
+		time.Sleep(timeout)
+		stop <- struct{}{}
```

Comment:
How do you notify to cleanup?  
It looks like it just hard-stops now.

Reply:
I should add cleanups to the stop case. I don't think they are super important here since that is the end of the program, but better to always cleanup rather than sometimes clean up.